confidential-assets/src/confidentialTransfer.ts

```diff
@@ -86,55 +112,75 @@ export class ConfidentialTransfer {
     this.auditorEncryptionKeys = args.auditorEncryptionKeys;
     this.auditorsU8EncryptionKeys = args.auditorsU8EncryptionKeys;
     this.auditorsCBList = args.auditorsCBList;
-    this.encryptedActualBalance = args.encryptedActualBalance;
+    this.encryptedSenderBalance = args.encryptedSenderBalance;
     this.confidentialAmountToTransfer = args.confidentialAmountToTransfer;
 
-    this.confidentialAmountAfterTransfer = args.confidentialAmountAfterTransfer;
+    this.confidentialBalanceAfterTransfer = args.confidentialAmountAfterTransfer;
     this.encryptedAmountByRecipient = args.encryptedAmountByRecipient;
-    this.randomness = args.randomness;
+    this.transferAmountRandomness = args.transferAmountRandomness;
     this.newBalanceRandomness = args.newBalanceRandomness;
+
+    this.encryptedAmountToTransfer = this.confidentialAmountToTransfer.getAmountEncrypted(
+      args.senderDecryptionKey.publicKey(),
+      this.transferAmountRandomness,
+    );
+    this.encryptedBalanceAfterTransfer = this.confidentialBalanceAfterTransfer.getAmountEncrypted(
+      args.senderDecryptionKey.publicKey(),
+      this.newBalanceRandomness,
+    );
   }
 
   static async create(args: CreateConfidentialTransferOpArgs) {
-    const randomness = args.randomness ?? ed25519GenListOfRandom(ConfidentialAmount.CHUNKS_COUNT);
+    const {
+      encryptedSenderBalance,
+      senderDecryptionKey,
+      recipientEncryptionKey,
+      amountToTransfer,
+      auditorEncryptionKeys = [],
+    } = args;
+    const transferAmountRandomness =
+      args.transferAmountRandomness ?? ed25519GenListOfRandom(ConfidentialAmount.CHUNKS_COUNT);
     const newBalanceRandomness = ed25519GenListOfRandom(ConfidentialAmount.CHUNKS_COUNT);
-    const recipientPublicKeyU8 = publicKeyToU8(args.recipientEncryptionKey);
+    const recipientPublicKeyU8 = publicKeyToU8(recipientEncryptionKey);
 
-    const confidentialAmountToTransfer = ConfidentialAmount.fromAmount(args.amountToTransfer, {
+    const confidentialAmountToTransfer = ConfidentialAmount.fromAmount(amountToTransfer, {
       chunksCount: ConfidentialAmount.CHUNKS_COUNT / 2,
     });
 
     const encryptedAmountByRecipient = confidentialAmountToTransfer.amountChunks.map((chunk, i) =>
```

Comment:
note this is a very unfortunate artifact of the way confidentialAmount does encrypting.  can be fixed after - https://github.com/aptos-labs/aptos-ts-sdk/pull/711